tart2ms/ms_helper.py

```diff
@@ -21,18 +29,29 @@
     logger.warning("Cannot import Africanus API. MODEL_DATA filling capabilities disabled")
     AFRICANUS_DFT_AVAIL = False
 
-def azel2radec(az, el, location, obstime):
+def azel2radec(az, el, location, obstime, distance=None):
     """ az, el -- in degrees
+        distance -- needed to convert solar system bodies back to J2000 in the barycentric frame
         location -- observer location on the ground 
         time -- astropy time for the observation
         returns radec in radians (fk5 frame)
     """
-    dir_altaz = SkyCoord(alt=el*u.deg, az=az*u.deg, obstime=obstime,
-                         frame='altaz', location=location)
-    dir_j2000 = dir_altaz.transform_to('fk5')
+    if distance is None:
+        dir_altaz = SkyCoord(alt=el*u.deg, az=az*u.deg, obstime=obstime,
+                            frame='altaz', location=location)
+    else:
+        dir_altaz = SkyCoord(alt=el*u.deg, az=az*u.deg, obstime=obstime,
+                            frame='altaz', location=location, distance=distance)
+    dir_j2000 = dir_altaz.transform_to('icrs')
```

Comment:
I realize for a solar system body it is necessary to do barycentric corrections when going from alt az to some epoch other than Jnow.

Reply:
Best to file an issue on this.